Python/pywarpx/picmi.py

```diff
@@ -1656,12 +1657,52 @@ def laser_antenna_initialize_inputs(self, laser):
 
 
 class LoadInitialField(picmistandard.PICMI_LoadGriddedField):
+    def init(self, kw):
+        self.do_divb_cleaning_external = kw.pop('warpx_do_divb_cleaning_external', True)
+
     def applied_field_initialize_inputs(self):
         pywarpx.warpx.read_fields_from_path = self.read_fields_from_path
         if self.load_E:
             pywarpx.warpx.E_ext_grid_init_style = 'read_from_file'
         if self.load_B:
             pywarpx.warpx.B_ext_grid_init_style = 'read_from_file'
+            pywarpx.warpx.do_divb_cleaning_external = True
+
+class LoadInitialFieldFromPython:
+    """
+    Field Initializer that takes a function handle to be registered as a callback.
+    The function is expected to write the E and/or B fields into the
+    fields.Bx/y/zFPExternalWrapper() multifab. The callback is installed
+    in the beforeInitEsolve hook. This should operate identically to laoding from
+    a file.
+
+    Parameters
+    ----------
+    warpx_do_divb_cleaning_external: bool, default=True
+        Flag that controls whether or not to execute the Projection based B-field divergence cleaner.
+
+    load_E: bool, default=True
+        E field is expected to be loaded in the registered callback.
+
+    load_B: bool, default=True
+        B field is expected to be loaded in the registered callback.
+    """
+    def __init__(self, **kw):
+        self.do_divb_cleaning_external = kw.pop('warpx_do_divb_cleaning_external', True)
```

Comment:
I'm wondering about the default `True` here and in `LoadInitialField`. Usually we keep these default to `None` so that no parameter is passed to WarpX and therefore the C++ defaults are used (instead of needing to work at maintaining the same default settings in picmi and in C++ separately). Is it needed to have this default to `True`?